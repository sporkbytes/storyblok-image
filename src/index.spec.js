import { transform } from './index';

const testImageUrl =
	'https://a.storyblok.com/f/39898/3310x2192/e4ec08624e/demo-image.jpeg';
const testImageWithFaceUrl =
	'https://a.storyblok.com/f/39898/2250x1500/c15735a73c/demo-image-human.jpeg';

describe('Base Functionality', () => {
	it('should change domains and focus on the center of the image when transforming with no options', () => {
		expect(transform(testImageUrl)).toMatchSnapshot();
	});
});

describe('Custom focal point', () => {
	it('should focus on the bottom of the image when focalPointY is above 50', () => {
		expect(
			transform(testImageWithFaceUrl, {
				focalPointY: 80,
				height: 130,
				width: 600,
			})
		).toMatchSnapshot();
	});

	it('should focus on the top of the image when focalPointY is below 50', () => {
		expect(
			transform(testImageWithFaceUrl, {
				focalPointY: 20,
				height: 130,
				width: 600,
			})
		).toMatchSnapshot();
	});
});

describe('Facial detection and smart cropping', () => {
	it('should focus on a face when the smartCrop option is set to true', () => {
		expect(
			transform(testImageWithFaceUrl, {
				height: 130,
				smartCrop: true,
				width: 600,
			})
		).toMatchSnapshot();
	});
});

describe('Resizing', () => {
	it('should resize the image when given height and width options', () => {
		expect(
			transform(testImageUrl, { height: 500, width: 500 })
		).toMatchSnapshot();
	});

	it('should resize proportional to width when given a width option and no height', () => {
		expect(transform(testImageUrl, { width: 200 })).toMatchSnapshot();
	});

	it('should resize proportional to height when given a height option and no width', () => {
		expect(transform(testImageUrl, { height: 200 })).toMatchSnapshot();
	});
});
