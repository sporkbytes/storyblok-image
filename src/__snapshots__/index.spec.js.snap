// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Functionality should change domains and focus on the center of the image when transforming with no options 1`] = `"https://img2.storyblok.com/0x0/filters:focal(1605x1046:1705x1146)/f/39898/3310x2192/e4ec08624e/demo-image.jpeg"`;

exports[`Custom focal point should focus on the bottom of the image when focalPointY is above 50 1`] = `"https://img2.storyblok.com/600x130/filters:focal(1075x1150:1175x1250)/f/39898/2250x1500/c15735a73c/demo-image-human.jpeg"`;

exports[`Custom focal point should focus on the top of the image when focalPointY is below 50 1`] = `"https://img2.storyblok.com/600x130/filters:focal(1075x250:1175x350)/f/39898/2250x1500/c15735a73c/demo-image-human.jpeg"`;

exports[`Facial detection and smart cropping should focus on a face when the smartCrop option is set to true 1`] = `"https://img2.storyblok.com/600x130/smart/f/39898/2250x1500/c15735a73c/demo-image-human.jpeg"`;

exports[`Resizing should resize proportional to height when given a height option and no width 1`] = `"https://img2.storyblok.com/0x200/filters:focal(1605x1046:1705x1146)/f/39898/3310x2192/e4ec08624e/demo-image.jpeg"`;

exports[`Resizing should resize proportional to width when given a width option and no height 1`] = `"https://img2.storyblok.com/200x0/filters:focal(1605x1046:1705x1146)/f/39898/3310x2192/e4ec08624e/demo-image.jpeg"`;

exports[`Resizing should resize the image when given height and width options 1`] = `"https://img2.storyblok.com/500x500/filters:focal(1605x1046:1705x1146)/f/39898/3310x2192/e4ec08624e/demo-image.jpeg"`;
